### PR TITLE
[NPU][Lora] Fix lora ascend accuracy

### PR DESCRIPTION
**Pull request by JiaruiChang5268**
* cherry #15912 :LoRA: Ascend: Update ascend LoRA backend to support new kernels

* Fix lora ascend accuracy

---------

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/srt/lora/backend/ascend_backend.py [hunk 1]
```diff
@@ -84,19 +82,15 @@ def run_qkv_lora(
...
         **kwargs,
     ) -> torch.Tensor:
-        num_slices = 3
         assert isinstance(qkv_lora_b, torch.Tensor)
 
         total_seq_len, _ = x.shape
-        _, weight_intermediate_dim, _ = qkv_lora_a.shape
-        _, weight_out_dim, _ = qkv_lora_b.shape
-        max_rank = weight_intermediate_dim // num_slices
+        _, weight_intermediate_dim, max_rank = qkv_lora_a.shape
```
Comment (vlserov) [inline python/sglang/srt/lora/backend/ascend_backend.py:92]: No need to add max_rank here, since it will be overwritten line below

File: python/sglang/srt/layers/quantization/unquant.py [hunk 2]
```diff
@@ -119,6 +120,8 @@ def create_weights(
     def process_weights_after_loading(self, layer: torch.nn.Module) -> None:
         if _is_cpu and _is_cpu_amx_available:
             _amx_process_weight_after_loading(layer, ["weight"])
+        if _is_npu:
+            layer.weight.data = npu_format_cast(layer.weight.data)
```
Comment (OrangeRedeng) [inline python/sglang/srt/layers/quantization/unquant.py]: Hello, could you please provide accuracy and benchmark tests (because in my earliar experiments this change reduced the perfomance)

File: python/sglang/srt/lora/backend/ascend_backend.py [hunk 3]
```diff
@@ -167,37 +153,31 @@ def run_gate_up_lora(
...
             lora_a_output,
-            1.0,
         )
 
-        scaling = (
-            self.batch_info.scalings.gather(0, self.batch_info.weight_indices)
-            .repeat_interleave(self.batch_info.seg_lens, output_size=total_seq_len)
-            .unsqueeze(-1)
+        lora_ranks = self.batch_info.lora_ranks.clone()
+        lora_ranks[1] = max_rank
```
Comment (vlserov) [inline python/sglang/srt/lora/backend/ascend_backend.py:171]: Since can be used more than one adapter, shouldn't be hardcoded,
Why lora_ranks  should be changed during execution?

File: python/sglang/srt/layers/quantization/unquant.py [hunk 4]
```diff
@@ -119,6 +120,8 @@ def create_weights(
     def process_weights_after_loading(self, layer: torch.nn.Module) -> None:
         if _is_cpu and _is_cpu_amx_available:
             _amx_process_weight_after_loading(layer, ["weight"])
+        if _is_npu:
```
Comment (shengzhaotian) [inline python/sglang/srt/layers/quantization/unquant.py:123]: Please check whether this modification is necessary. If yes, add an isolation measure to avoid affecting other linear components.

File: python/sglang/srt/server_args.py [hunk 5]
```diff
@@ -4901,7 +4901,7 @@ def check_lora_server_args(self):
 
         if self.enable_lora:
             # Validate compatibility with speculative decoding
-            if self.speculative_algorithm not in ["NGRAM", None]:
+            if self.speculative_algorithm not in ["NGRAM", "SUFFIX", None]:
```
Comment (shengzhaotian) [inline python/sglang/srt/server_args.py:4904]: This change seems to be irrelevant to this PR. Please confirm.
